### PR TITLE
fix: rfox invalidate queries on Tx confirm

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Fixes one of two hard things in software engineering, by waiting for the Txs to be completed before invalidating queries.
Implemented using a new `onTxConfirmed` callback prop in the spirit of status component abstraction.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/7015
- contributes to https://github.com/shapeshift/web/issues/6964

Low - ensure the refactored flow doesn't result in errors being thrown when consuming `useQuery()` (or wagmi's wrappers) to get the queryKeys.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Proceed with stake, unstake, and change address Txs
- Ensure the previous values (i.e staked amount, share of pool...) are correctly invalidated when coming back to input/confirm after Tx confirm and we don't have any stale values.

### Engineering

- ^

### Operations

- ^

## Screenshots (if applicable)

https://jam.dev/c/70f352f0-dd38-4e10-990a-638b7585ea62

### CODE REVIEW

**Review by NeOMakinG — APPROVED** [review]
https://jam.dev/c/1a2cd6c6-84e9-475d-8015-578544b274e7

Staking is happy, informations are updated as expected

Unstaking is happy, information is updated as expected and pool share looks fine.

Another staking pool share is also happy and display the right amount, 10 - 7 + 7 actually displays 

image

Changing address is happy too, the displayed address is updated after using the change address feature

File: src/pages/RFOX/components/ChangeAddress/ChangeAddressStatus.tsx [hunk 1]
```diff
@@ -50,6 +52,12 @@ export const ChangeAddressStatus: React.FC<ChangeAddressRouteProps & ChangeAddre
   )
   const tx = useAppSelector(state => selectTxById(state, serializedTxIndex))
 
+  useEffect(() => {
+    if (tx?.status !== TxStatus.Confirmed) return
+
+    handleTxConfirmed()
+  }, [handleTxConfirmed, tx?.status])
+
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply src/pages/RFOX/components/ChangeAddress/ChangeAddressStatus.tsx:60]: a bit more details here to be discussed: https://github.com/shapeshift/web/issues/7034

[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/pages/RFOX/components/ChangeAddress/ChangeAddressStatus.tsx:60]: Good insights and thanks for the `useTxStatus` issue @NeOMakinG, v. good one indeed!

As stated in the PR description discussed, this PR just moves code around - basically invalidating on Tx confirmed vs. on broadcast currently.
Now any component could do the whole `serializedTxIndex` and `tx.status` check then invalidate so long as it's mounted when the Tx confirms - which excludes input and confirm, and effectively leaves us with status, which already has the `Tx` handy for us to do the checks in an effect. Or we could obviously do that in `<ChangeAddressRoutes />` and its stake and unstake frens, but then we start bringing in way more boilerplate in for the sole purpose of invalidating queries.

If you'd want to tackle #7034, I'm all for it - agree would be a nice hook to add in and avoid duplicating those bits

File: src/pages/RFOX/components/Stake/StakeStatus.tsx [hunk 2]
```diff
@@ -54,6 +58,12 @@ export const StakeStatus: React.FC<StakeRouteProps & StakeStatusProps> = ({
 
   const tx = useAppSelector(state => selectTxById(state, serializedTxIndex))
 
+  useEffect(() => {
+    if (tx?.status !== TxStatus.Confirmed) return
+
+    handleTxConfirmed()
+  }, [handleTxConfirmed, queryClient, tx?.status])
```
Comment (NeOMakinG) [inline src/pages/RFOX/components/Stake/StakeStatus.tsx:65]: Is there a reason why we have the queryClient as a dependency? And why don't we have it on `ChangeAddressStatus` side

Comment (gomesalexandre) [reply src/pages/RFOX/components/Stake/StakeStatus.tsx]: Nice catch - 557045eddb. The issue here isn't so much we have it as a dep but rather we shouldn't have `queryClient` in this component anymore altogether.

**Pull request merged**